### PR TITLE
New Akima spline component for interpolation.

### PR DESCRIPTION
**Pull request by Kenneth-T-Moore**

### CODE REVIEW

File: openmdao/components/akima_spline_comp.py [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+        Declare options.
+        """
+        self.options.declare('num_control_points', types=int, default=10,
+                             desc="Number of control points.")
+        self.options.declare('num_points', types=int, default=20,
+                             desc="Number of interpolated points.")
+        self.options.declare('vec_size', types=int, default=1,
+                             desc='Number of independent rows to interpolate.')
+        self.options.declare('name', types=str, default=None, allow_none=True,
+                             desc="Name to use for the interpolated variable.")
```
Comment (swryan) [inline openmdao/components/akima_spline_comp.py]: can 'name' really be None?  It looks like it's used to create the input and output vars...

File: openmdao/docs/features/building_blocks/components/akima_spline_comp.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    AkimaSplineComp
+    options
+
+
+AkimaSplineComp Basic Example
+-----------------------------
+
+The following is a simple example where an AkimaSplineComp is used to approximate a curve that has
+11 points where we would like to evaluate it.  The approximating curve contains 6 points. Note that
+unlke the `BsplinesComp`, the control points fall on the curve.
```
[the remark this answers is not in the document]
Comment (Kenneth-T-Moore) [reply openmdao/docs/features/building_blocks/components/akima_spline_comp.rst]: I think I'd rather rename the Bsplines Comp, but I am not sure.  Can you guys have Tristan weigh in on the proper name for this component?

File: openmdao/components/akima_spline_comp.py [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+                             desc="When True, the x control point grid is a component input.")
+        self.options.declare('units', types=string_types, default=None, allow_none=True,
+                             desc="Units to use for the y and ycp variables.")
+        self.options.declare('x_units', types=string_types, default=None, allow_none=True,
+                             desc="Units to use for the x and xcp variables.")
+        self.options.declare('delta_x', default=0.1,
+                             desc="half-width of the smoothing interval added in the valley of "
+                             "absolute-value function. This allows the derivatives with respect to"
+                             " the data points (dydxpt, dydypt) to also be C1 continuous. Set to "
+                             "parameter to 0 to get the original Akima function (but only if you "
```
[the remark this answers is not in the document]
Comment (Kenneth-T-Moore) [reply openmdao/components/akima_spline_comp.py]: fixed

File: openmdao/components/akima_spline_comp.py [hunk 4]
```diff
@@ -0,0 +1,460 @@
...
+                             "don't need dydxpt, dydypt")
+        self.options.declare('eval_at', default='end', values=['end', 'cell_center'],
+                             desc="Where the return values are evaluate on the spline. When set "
+                             "to end, compute num_points values spanning the full interval set by "
+                             "the training points. When set to 'cell_center', compute values at "
+                             "centerpoints of num_points cells. Only used if xcp_name is not "
+                             "given.")
```
[the remark this answers is not in the document]
Comment (Kenneth-T-Moore) [reply openmdao/components/akima_spline_comp.py]: Fixed. Leftover from the old API.

File: openmdao/components/akima_spline_comp.py (hunk 1 above)
Comment (Kenneth-T-Moore) [reply openmdao/components/akima_spline_comp.py]: Fixed.  Fixed all but question about what to call the component.

**Pull request merged**